ResearchKit/Common/ORKDiscreteGraphView.m

```diff
@@ -0,0 +1,1023 @@
+ specific prior written permission. No license is granted to the trademarks of 
+ the copyright holders even if such marks are included in this software. 
+ 
+ THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" 
+ AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE 
+ IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+ ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE 
+ FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL 
+ DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR 
+ SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER 
+ CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, 
+ OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE 
+ OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. 
+*/
+
+ 
+#import "ORKDiscreteGraphView.h"
+#import <ResearchKit/ORKAxisView.h>
+#import <ResearchKit/ORKCircleView.h>
+
+
+NSString * const kORKDiscreteGraphViewTriggerAnimationsNotification = @"ORKDiscreteGraphViewTriggerAnimationsNotification";
+NSString * const kORKDiscreteGraphViewRefreshNotification = @"ORKDiscreteGraphViewRefreshNotification";
+
+static CGFloat const kYAxisPaddingFactor = 0.15f;
+static CGFloat const kORKGraphLeftPadding = 10.f;
+static CGFloat const kAxisMarkingRulerLength = 8.0f;
+
+static NSString * const kFadeAnimationKey = @"LayerFadeAnimation";
+static NSString * const kGrowAnimationKey = @"LayerGrowAnimation";
+
+static CGFloat const kFadeAnimationDuration = 0.2;
+static CGFloat const kGrowAnimationDuration = 0.1;
+static CGFloat const kPopAnimationDuration  = 0.3;
+
+static CGFloat const kSnappingClosenessFactor = 0.3f;
+
+@interface ORKDiscreteGraphView ()
+
+@property (nonatomic, strong) NSMutableArray *dataPoints;//actual data
+
+@property (nonatomic, strong) NSMutableArray *xAxisPoints;
+
+@property (nonatomic, strong) NSMutableArray *yAxisPoints;//normalised for this view
+
+@property (nonatomic, strong) UIView *plotsView; //Holds the plots
+
+@property (nonatomic, strong) ORKAxisView *xAxisView;
+
+@property (nonatomic, strong) UIView *yAxisView;
+
+@property (nonatomic, strong) UILabel *emptyLabel;
+
+@property (nonatomic) BOOL hasDataPoint;
+
+@property (nonatomic, strong) UIView *scrubberLine;
+
+@property (nonatomic, strong) UILabel *scrubberLabel;
+
+@property (nonatomic, strong) UIView *scrubberThumbView;
+
+@property (nonatomic, readwrite) CGFloat minimumValue;
+
+@property (nonatomic, readwrite) CGFloat maximumValue;
+
+@property (nonatomic, strong) NSMutableArray *xAxisTitles;
+
+@property (nonatomic) NSInteger numberOfXAxisTitles;
+
+@property (nonatomic, strong) NSMutableArray *referenceLines;
+
+@property (nonatomic, strong) NSMutableArray *pathLines;
+
+@property (nonatomic, strong) NSMutableArray *dots;
+
+@property (nonatomic) BOOL shouldAnimate;
```

Comment:
Move common properties to `ORKBaseGraphView`.

Preserve the comments but beautify their formatting (e.g., `+@property (nonatomic, strong) NSMutableArray *dataPoints; // Actual data`)